backend/src/main/java/de/jankahle/capstone/service/RecipeService.java

```diff
@@ -31,7 +31,9 @@ public RecipeService(RecipeMongoDB recipeMongoDB, ImageReaderService imageReader
 
     public Recipe saveRecipe(RecipeDto recipeDto) {
         Recipe recipe = recipeDto.toRecipe();
-        recipe.setId(IdUtility.generateId());
+        if (recipe.getId().isEmpty()) {
+            recipe.setId(IdUtility.generateId());
+        }
```

Comment:
`(...)MongoDb`: In order not to have to reflect future changes, i would name the db neutrally. The pagingundsorting interface is supposed to hide that.